src/libstd/sys/unix/fs.rs

```diff
@@ -49,11 +49,11 @@ pub struct FileAttr {
 }
 
 pub struct ReadDir {
-    dirp: Dir,
+    dirp: Option<Dir>,
```

Comment:
Is this change related to the infinite loop issue?

Reply:
Yes. `Option` is used to store whether we are at the end of the directory stream or not.